src/prof_log.c

```diff
@@ -673,36 +672,26 @@ prof_log_stop(tsdn_t *tsdn) {
 #undef PROF_LOG_STOP_BUFSIZE
 
 bool prof_log_init(tsd_t *tsd) {
-	if (opt_prof_log) {
-		prof_log_start(tsd_tsdn(tsd), NULL);
+	if (malloc_mutex_init(&log_mtx, "prof_log",
+	    WITNESS_RANK_PROF_LOG, malloc_mutex_rank_exclusive)) {
+		return true;
+	}
+
+	if (!opt_prof_log) {
+		return false;
 	}
 
+	prof_log_start(tsd_tsdn(tsd), NULL);
+
 	if (atexit(prof_log_stop_final) != 0) {
 		malloc_write("<jemalloc>: Error in atexit() "
 			     "for logging\n");
 		if (opt_abort) {
 			abort();
 		}
-	}
-
-	if (malloc_mutex_init(&log_mtx, "prof_log",
-	    WITNESS_RANK_PROF_LOG, malloc_mutex_rank_exclusive)) {
-		return true;
-	}
-
-	if (ckh_new(tsd, &log_bt_node_set, PROF_CKH_MINITEMS,
```

Comment:
Do we just not need to initialize these hash tables? How come?

Reply:
We can initialize them here but we don't have to. `prof_try_log()` checks about these tables and initializes them if needed - and it has to do so because there can be multiple segmented logging period. I think it's a good idea in itself to delay the table initialization until the first freed sample comes in, which is I believe also the intention of the original design (which I'm still learning via reading the code).